lib/phoenix/router/route.ex

```diff
@@ -43,20 +47,28 @@ defmodule Phoenix.Router.Route do
   Builds the expressions used by the route.
   """
   def exprs(route) do
-    {path, binding} = build_path_and_binding(route.path)
+    {path, binding} = build_path_and_binding(route)
 
     %{path: path,
       host: build_host(route.host),
+      verb_match: verb_match(route.verb),
       binding: binding,
       dispatch: build_dispatch(route, binding)}
   end
 
-  defp build_path_and_binding(path) do
-    {params, segments} = Plug.Router.Utils.build_path_match(path)
+  defp verb_match(:*), do: Macro.var(:_verb, nil)
+  defp verb_match(verb), do: verb |> to_string() |> String.upcase()
+
+  defp build_path_and_binding(%Route{path: path} = route) do
+    {params, segments} = case route.kind do
+      :forward -> Plug.Router.Utils.build_path_match(path <> "/*_forward_path_info")
+      :match   -> Plug.Router.Utils.build_path_match(path)
+    end
 
-    binding = Enum.map(params, fn var ->
-      {Atom.to_string(var), Macro.var(var, nil)}
-    end)
+    binding =
+      params
```

Comment:
Btw, this could be cleaner as a for-comprehension or a List.delete (it doesn't matter though).